client/src/sections/courses/LearningSection/AddLearnerModal.tsx

```diff
@@ -1,125 +1,184 @@
+/* eslint-disable multiline-ternary */
+import React, { useState } from 'react';
 import Button from '@/src/shared/components/Button';
 import Modal from '@/src/shared/components/Modal/Modal';
 import Pagination from '@/src/shared/components/Pagination';
 import SearchBar from '@/src/shared/components/SearchBar/SearchBar';
 import CloseIcon from '@/src/shared/icons/CloseIcon';
-import { type Learner } from '@/src/shared/utils';
-import React, { useState } from 'react';
+import { useRouter } from 'next/router';
+import { useSelector } from 'react-redux';
+import { useEnrollLearnerMutation, useGetLearnerQuery } from '@/services/traineeAPI';
+import { type Trainee, addTrainees, toggleLearner } from '@/features/course/learnerSlice';
+import { useAppDispatch } from '@/app/hooks';
+import { type RootState } from '@/app/store';
 
-interface AddLearnerModalProps {
-  learners: Learner[];
-}
-
-const AddLearnerModal: React.FC<AddLearnerModalProps> = ({ learners }: AddLearnerModalProps): JSX.Element => {
-  const [currentPage, setCurrentPage] = useState(1);
+const AddLearnerModal: React.FC = (): JSX.Element => {
+  const router = useRouter();
+  const params = router.query;
   const [isModalOpen, setIsModalOpen] = useState(false);
+  const [search, setSearch] = useState('');
+  const [page, setPage] = useState(1);
+  const [selectedIds, setSelectedIds] = useState<number[]>([]);
+  const [selectedEnrollees, setSelectedEnrollees] = useState<Trainee[]>([]);
+  const courseID = params.id;
+  const dispatch = useAppDispatch();
 
-  const handleAddLearners = (): void => {
-    alert('Add Learners');
-  };
+  const { isEnrollee } = useSelector((state: RootState) => state.courseLearner);
```

Comment:
```suggestion
  const { isEnrollee } = useAppSelector((state) => state.courseLearner);
```

Please use useAppSelector instead, it already has RootState type assigned, so you do not need to import and assign it in other places

Please apply all related